provider/server/machine_annotations_update_test.go

```diff
@@ -0,0 +1,90 @@
+
+package server_test
+
+import (
+	"github.com/digitalocean/go-libvirt"
+	iri "github.com/ironcore-dev/ironcore/iri/apis/machine/v1alpha1"
+	irimeta "github.com/ironcore-dev/ironcore/iri/apis/meta/v1alpha1"
+	libvirtutils "github.com/ironcore-dev/libvirt-provider/pkg/libvirt/utils"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+)
+
+var _ = Describe("UpdateMachineAnnotations", func() {
+	It("should update machine annotations", func(ctx SpecContext) {
+		ignitionData := []byte("urjhikmnbdjfkknhhdddeee")
+		By("creating a machine")
+		createResp, err := machineClient.CreateMachine(ctx, &iri.CreateMachineRequest{
+			Machine: &iri.Machine{
+				Metadata: &irimeta.ObjectMetadata{
+					Labels: map[string]string{
+						"machinepoolletv1alpha1.MachineUIDLabel": "foolabel",
+					},
+				},
+				Spec: &iri.MachineSpec{
+					Power:             iri.Power_POWER_ON,
+					Class:             machineClassx3xlarge,
+					IgnitionData:      ignitionData,
+					Volumes:           nil,
+					NetworkInterfaces: nil,
+				},
+			},
+		})
+		Expect(err).NotTo(HaveOccurred())
+		Expect(createResp).NotTo(BeNil())
+
+		DeferCleanup(func(ctx SpecContext) {
+			Eventually(func() bool {
+				_, err := machineClient.DeleteMachine(ctx, &iri.DeleteMachineRequest{MachineId: createResp.Machine.Metadata.Id})
+				Expect(err).ShouldNot(HaveOccurred())
+				_, err = libvirtConn.DomainLookupByUUID(libvirtutils.UUIDStringToBytes(createResp.Machine.Metadata.Id))
+				return libvirt.IsNotFound(err)
+			}).Should(BeTrue())
+		})
+
+		By("ensuring domain and domain XML is created for machine")
+		var domain libvirt.Domain
+		Eventually(func() error {
+			domain, err = libvirtConn.DomainLookupByUUID(libvirtutils.UUIDStringToBytes(createResp.Machine.Metadata.Id))
+			return err
+		}).Should(Succeed())
+		domainXMLData, err := libvirtConn.DomainGetXMLDesc(domain, 0)
+		Expect(err).NotTo(HaveOccurred())
+		Expect(domainXMLData).NotTo(BeEmpty())
+
+		By("ensuring domain for machine is in running state")
+		Eventually(func() libvirt.DomainState {
+			domainState, _, err := libvirtConn.DomainGetState(domain, 0)
+			Expect(err).NotTo(HaveOccurred())
+			return libvirt.DomainState(domainState)
+		}).Should(Equal(libvirt.DomainRunning))
+
+		By("updating machine annotations")
+		_, err = machineClient.UpdateMachineAnnotations(ctx, &iri.UpdateMachineAnnotationsRequest{
+			MachineId: createResp.Machine.Metadata.Id,
+			Annotations: map[string]string{
+				"machinepoolletv1alpha1.MachineUIDLabel": "fooUpdatedAnnotation",
+			},
+		})
+		Expect(err).NotTo(HaveOccurred())
+
+		By("ensuring correct annotations")
+		Eventually(func() *irimeta.ObjectMetadata {
+			listResp, err := machineClient.ListMachines(ctx, &iri.ListMachinesRequest{
+				Filter: &iri.MachineFilter{
+					Id: createResp.Machine.Metadata.Id,
+				},
+			})
+			Expect(err).NotTo(HaveOccurred())
+			Expect(listResp.Machines).NotTo(BeEmpty())
+			Expect(len(listResp.Machines)).Should(Equal(1))
```

Comment:
```suggestion
			Expect(listResp).Should(HaveLen(1))
```

Reply:
Replaced with Expect(...).Should(HaveLen()) wherever applicable